lib/crypto.js

```diff
@@ -83,6 +83,7 @@ exports.createHmac = exports.Hmac = Hmac;
 function Hmac(hmac, key, options) {
   if (!(this instanceof Hmac))
     return new Hmac(hmac, key, options);
+  this.key = key;
```

Comment:
What are the secondary security consequences of adding the secret key as an instance property in the constructor?